pkg/encoding/encoding.go

```diff
@@ -165,18 +164,6 @@ func tryFindJson(in []byte) (*json.RawMessage, bool) {
 	return nil, false
 }
 
-// DecodeSummary decodes the TypeMeta, ContentEncoding and ContentType fields from the 'Unknown'
-// protobuf envelope of the given storage data.
-func DecodeSummary(inMediaType string, in []byte, out io.Writer) error {
```

Comment:
Can you help me understand why removing this function better? While `DecodeSummary` may not do much, it make sure there is in consistency in printing out summary for batch and non-batch mode. With this change, whenever there is some new meta field we want to display, the change has to be in 2 places, and that would be more error prone.

Reply:
I think this function should only be used by cmd without exception, or just move it to cmd 😄.

Reply:
I really don't see much benefits of this refactoring. `DecodeSummary` is closely related to decode function itself, and can be used elsewhere in theory. 
@jmhbnz WDYT?

Reply:
It's minor pr, though for me it's not particularly clear which high level problem this pr is trying to work towards which is why I have not already added review or approval for it.

I suggest we try focus community effort on progressing towards an agreed direction rather than refactor for the sake of it. Our challenge is obviously that auger has only recently come into sig-etcd and we have not created issues yet to establish direction to go in.

Perhaps we should step back a little from this code diff and try to get agreement on what problems we are wanting to solve so it's clear for future pull requests if they align with the direction we are trying to go in?

If project members or contributors can propose issues to start discussion it would be very helpful.

Reply:
I agree with you. I've already made issue #3 describing what I wanted to achieve. which is directly edit the data from etcd, and using it just like `kwokctl`.

Since the code changes are already a bit much, I'm trying to split it up as much as I can for review.

Reply:
Thanks - For me there is no clear link between that issue and this pr.  Why would this pr be required for #3?

Reply:
I'm trying to split it up as much for review. 
but if you have time to review a big PR, I'll do it next week.

Reply:
The purpose of this PR is to expose `encoding.DecodeTypeMeta`, make it can be used in other packages.
then see that `encoding.DecodeSummary` only applies to cmd, so it moves over to cmd

Reply:
In the interest of getting this merged I suggest we limit the pr change to just making `DecodeTypeMeta` public. I am not in favor of the change to `DecodeSummary`.

Reply:
Done. change it as you say.

> I am not in favor of the change to DecodeSummary.

Can you tell me why? As I understand `DecodeSummary ` will never be used by others.

Reply:
Personally because it is so early in lifecycle of auger within sig-etcd. We don't yet have regular releases occurring, or announcements of deprecations, or clear understandings of all use cases external to the project. For these reasons I am wary of removing a previously public function just for a very optional refactor.

Reply:
Thanks for the clarification, I finally understand the difference in how we think about this, you see it as a legacy project that needs to ensure compatibility, and I see it as a new project, the history is actually in [github.com/jpbetz/auger](https://github.com/jpbetz/auger).